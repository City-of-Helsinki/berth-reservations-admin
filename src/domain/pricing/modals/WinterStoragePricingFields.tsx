import React, { FunctionComponent } from 'react';
import { TextInput } from 'hds-react/lib';

import styles from './modals.module.scss';
import Grid from '../../../common/grid/Grid';
import Select from '../../../common/select/Select';
import { HarborService, WinterStoragePrice } from '../PricingPage';

export interface WinterStoragePricingFieldsProps {
  test?: 'test';
  initialValues: WinterStoragePrice;
}

const WinterStoragePricingFields: FunctionComponent<WinterStoragePricingFieldsProps> = () => {
  return (
    <>
      <div className={styles.row}>
        <TextInput
          id="baz"
          labelText="Tietue"
          value="Talvisäilytyspaikka"
          disabled
        />
      </div>
      <hr />
      <Grid colsCount={1} className={styles.row}>
        <Select
          labelText="Alue"
          value={'Kaisaniemi'}
          options={['Kaisaniemi'].map(option => {
            return { value: option, label: option };
          })}
          onChange={() => console.log('Test')}
        />
      </Grid>
      <Grid colsCount={2} className={styles.row}>
        <TextInput id="foo" labelText="Yksityinen (€)" value="116,00" />
        <TextInput id="bar" labelText="Yritys (€)" value="236.00" />
      </Grid>
      <Grid colsCount={2} className={styles.row}>
        <Select
          labelText="Aika"
          value={'Kausi'}
          options={[{ value: 'Kausi', label: 'Kausi' }]}
          onChange={() => console.log('Test')}
        />
      </Grid>
    </>
  );
};

export default WinterStoragePricingFields;
