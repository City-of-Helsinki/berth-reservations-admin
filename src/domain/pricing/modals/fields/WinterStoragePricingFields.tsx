import React, { FunctionComponent } from 'react';
import { TextInput } from 'hds-react/lib';
import { Field } from 'formik';
import { useTranslation } from 'react-i18next';

import styles from '../modals.module.scss';
import Grid from '../../../../common/grid/Grid';
import Select from '../../../../common/select/Select';
import FormTypeTitle from '../FormTypeTitle';

const WinterStoragePricingFields: FunctionComponent = () => {
  const { t } = useTranslation();

  const areaOptions = ['Kaisaniemi'];
  const periodOptions = ['season', 'month', 'year'];

  return (
    <>
      <FormTypeTitle
        label={t('common.terminology.dataEntry')}
        value={t('pricing.winterStorage.title')}
      />
      <hr />
      <Grid colsCount={2} className={styles.row}>
        <Field
          required={true}
          as={Select}
          id="area"
          name="area"
          labelText={t('pricing.winterStorage.area')}
          options={areaOptions.map(option => ({
            value: option,
            label: option,
          }))}
        />
      </Grid>
      <Grid colsCount={2} className={styles.row}>
        <Field
          required={true}
          as={TextInput}
          id="privateCustomer"
          name="privateCustomer"
          labelText={`${t('pricing.winterStorage.privateCustomer')} (€)`}
        />
        <Field
          required={true}
          as={TextInput}
          id="company"
          name="company"
          labelText={`${t('pricing.winterStorage.company')} (€)`}
        />
      </Grid>
      <Grid colsCount={2} className={styles.row}>
        <Field
          required={true}
          as={Select}
          id="period"
          labelText={t('pricing.winterStorage.period')}
          options={periodOptions.map(option => ({
            value: option,
            label: t([`common.periodTypes.${option}`]),
          }))}
        />
      </Grid>
    </>
  );
};

export default WinterStoragePricingFields;
