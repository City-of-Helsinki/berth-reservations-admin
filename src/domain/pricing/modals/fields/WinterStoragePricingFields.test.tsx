import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Form, Formik } from 'formik';

import WinterStoragePricingFields from './WinterStoragePricingFields';

const mockProps = {
  initialValues: {
    id: '1',
    area: 'Kaisaniemi',
    privateCustomer: 8.5,
    company: 17,
    period: 'season',
  },
};

describe('WinterStoragePricingFields', () => {
  const getWrapper = (props: { initialValues: any }): ShallowWrapper<any> =>
    shallow(
      <Formik
        initialValues={props.initialValues}
        onSubmit={jest.fn()}
        validate={jest.fn()}
      >
        {() => (
          <Form>
            <WinterStoragePricingFields />
          </Form>
        )}
      </Formik>
    );

  it('renders normally', () => {
    const wrapper = getWrapper(mockProps);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
