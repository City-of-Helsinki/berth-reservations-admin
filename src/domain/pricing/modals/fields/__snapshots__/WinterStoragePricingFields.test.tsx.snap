// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinterStoragePricingFields renders normally 1`] = `"<form action=\\"#\\"><div class=\\"row\\"><div class=\\"formTitle\\"><div class=\\"formTitleLabel\\">Tietue</div><div class=\\"formTitleValue\\">Talvisäilytyspaikka</div></div></div><hr/><div class=\\"grid cols2 row\\"><label><span class=\\"labelText\\">Alue</span><select name=\\"area\\" id=\\"area\\" class=\\"select\\"><option selected=\\"\\" value=\\"Kaisaniemi\\">Kaisaniemi</option></select></label></div><div class=\\"grid cols2 row\\"><div class=\\"_1cealf9APZDUwwgivaltwb\\"><label for=\\"privateCustomer\\" class=\\"_1NkZD_iEqeTFRGBnfzQ4zX _195GUCoLSCjwq2j1VtKb93 _2tInY8Y3RXDANhP6_8Ipb_ \\">Yksityinen (€)</label><div class=\\"_1ru3hdCZ3qhTQ_EVvRwBIh\\"><input type=\\"text\\" class=\\"_35GDyiOFLPVMWYFM4_BMWV CQwbnf7hlQrao8Za4G1Wo\\" id=\\"privateCustomer\\" placeholder=\\"\\" value=\\"8.5\\"/></div></div><div class=\\"_1cealf9APZDUwwgivaltwb\\"><label for=\\"company\\" class=\\"_1NkZD_iEqeTFRGBnfzQ4zX _195GUCoLSCjwq2j1VtKb93 _2tInY8Y3RXDANhP6_8Ipb_ \\">Yritys (€)</label><div class=\\"_1ru3hdCZ3qhTQ_EVvRwBIh\\"><input type=\\"text\\" class=\\"_35GDyiOFLPVMWYFM4_BMWV CQwbnf7hlQrao8Za4G1Wo\\" id=\\"company\\" placeholder=\\"\\" value=\\"17\\"/></div></div></div><div class=\\"grid cols2 row\\"><label><span class=\\"labelText\\">Aika</span><select id=\\"period\\" class=\\"select\\"><option value=\\"season\\">Kausi</option><option value=\\"month\\">1 kk</option><option value=\\"year\\">Vuosi</option></select></label></div></form>"`;
